src/coreclr/jit/gentree.cpp

```diff
@@ -3486,6 +3486,7 @@ unsigned Compiler::gtSetEvalOrder(GenTree* tree)
                 }
 #elif defined(TARGET_ARM64)
                 if ((*((__int64*)&(tree->AsDblCon()->gtDconVal)) == 0) ||
```

Comment:
This check should probably be changed to the new `IsFPPositiveZero` you added (or w/e the name is)